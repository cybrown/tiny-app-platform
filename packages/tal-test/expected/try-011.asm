main()
  entry:
    FunctionRef       name: toto1_0
    DeclareLocal      name: toto1, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    FunctionRef       name: toto2_1
    DeclareLocal      name: toto2, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal           0
    MakeArray
    Literal           0
    MakeRecord
    Local             toto2
    Call
toto1_0()
  entry:
    Local             missing_variable
    MakeArrayForBlock count: 1
toto2_1()
  entry:
    Try               catchLabel: try_catch_1, endTryLabel: try_end_2
    ScopeEnter
    Literal           0
    MakeArray
    Literal           0
    MakeRecord
    Local             toto1
    Call
    ScopeLeave        inBlock: false, count: 1
    TryPop
    Jump              try_end_2
  try_catch_1:
    ScopeEnter
    Local             errorToCatch
    ScopeLeave        inBlock: false
    Jump              try_end_2
  try_end_2:
    Pop               inBlock: false
    ScopeEnter
    Literal           "value from toto2"
    ScopeLeave        inBlock: false, count: 1
    MakeArrayForBlock count: 2
