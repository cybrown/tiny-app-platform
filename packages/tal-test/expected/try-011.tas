fun toto1() { missing_variable }

fun toto2() {
  try { toto1() } catch (errorToCatch)
  { "value from toto2" }
}

toto2()
