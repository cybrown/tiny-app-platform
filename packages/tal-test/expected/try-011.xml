<program>
  <declarelocal>
    <function>
      <block>
        <local />
      </block>
    </function>
  </declarelocal>
  <declarelocal>
    <function>
      <block>
        <try>
          <block>
            <call>
              <local />
            </call>
          </block>
          <nested>
            <local />
          </nested>
        </try>
        <block>
          <literal />
        </block>
      </block>
    </function>
  </declarelocal>
  <call>
    <local />
  </call>
</program>
